### PR TITLE
feat(HACBS-565): add cleanup-workspace task

### PR DESCRIPTION
**Pull request by johnbieren**
This commits add a task to cleanup a given directory that
is contained in the provided tekton workspace.

Signed-off-by: Johnny Bieren <jbieren@redhat.com>

### CODE REVIEW

File: definitions/cleanup-workspace/cleanup-workspace.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    - name: cleanup
+      image: quay.io/hacbs-release/release-utils
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        CLEANUP_DIR="$(workspaces.input.path)/$(params.subdirectory)"
+
+        if [ -d "${CLEANUP_DIR}" ] ; then
+          # Delete and directory and its contents
```
Comment (davidmogar) [inline definitions/cleanup-workspace/cleanup-workspace.yaml:24]:
```suggestion
          # Delete an directory and its contents
```

File: definitions/cleanup-workspace/cleanup-workspace.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    - name: input
+      description: Workspace where the directory to cleanup exists
+  steps:
+    - name: cleanup
+      image: quay.io/hacbs-release/release-utils
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        CLEANUP_DIR="$(workspaces.input.path)/$(params.subdirectory)"
```
Comment (davidmogar) [inline definitions/cleanup-workspace/cleanup-workspace.yaml:21]: Is it possible to pass an empty string as `subdirectory`? I have no idea tbh but, if so, we need a guard.

Comment (johnbieren) [reply definitions/cleanup-workspace/cleanup-workspace.yaml:26]: Yes, it let me pass the empty string

File: definitions/cleanup-workspace/cleanup-workspace.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  steps:
+    - name: cleanup
+      image: quay.io/hacbs-release/release-utils
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        if [ -z "$(params.subdirectory)" ] ; then
+            echo "The empty string is not a valid subdirectory"
+            exit 1
```
Comment (davidmogar) [inline definitions/cleanup-workspace/cleanup-workspace.yaml:23]: What about not breaking here and just show a warning like you do?

Comment (johnbieren) [reply definitions/cleanup-workspace/cleanup-workspace.yaml:23]: Why would we want to try to let `rm -rf ${mounted-workspace}` execute?

Comment (davidmogar) [reply definitions/cleanup-workspace/cleanup-workspace.yaml:23]: I meant to change it to `exit 0` with just a warning.

**Review by davidmogar — APPROVED** [review]

**Pull request merged**